R/mutate_super_vars.R

```diff
@@ -0,0 +1,210 @@
+  }
+  if (age_based_cap){
+    if (!any("age_range_description" == names(.sample.file))){
+      
+      # Following step to avoid having to require taxstats being attached (installed)
+      # to enjoy decoding of age variable.  A bit strange (as it is very likely that
+      # the user, who must have a sample file, would be using taxstats as well).
+      if (!isTRUE(requireNamespace("taxstats")) && !exists("age_range_decoder")){
+        age_range_decoder <- 
+          structure(list(age_range = c(11L, 10L, 9L, 8L, 7L, 6L, 5L, 4L, 3L, 2L, 1L, 0L), 
+                         age_range_description = structure(1:12, 
+                                                           .Label = c("under 20",  "20 to 24", "25 to 29", "30 to 34", "35 to 39", "40 to 44", "45 to 49",  "50 to 54", "55 to 59", "60 to 64", "65 to 69", "70 and over"), 
+                                                           class = c("ordered", "factor"))), 
+                    .Names = c("age_range", "age_range_description"), 
+                    class = c("data.frame", "data.table"), 
+                    row.names = c(NA, -12L))
+        
+      }
+      .sample.file <- merge(.sample.file, age_range_decoder, by = "age_range")
+    }
+    
+    # if an age falls between an age group, we assume ambiguously aged individuals are NOT entitled to cap2.
+    if (cap2_age < 65 && cap2_age >= 20){
+      cap2_age_group <- paste0(ceiling(cap2_age / 5) * 5, " to ", ceiling(cap2_age / 5) * 5 + 4)
+    } else {
+      if (cap2_age >= 65){
+        cap2_age_group <- "70 and over"
+      } else {
+        cap2_age_group <- "under 20"
+      }
+    }
+    
+    .sample.file[ , concessional_cap := ifelse(age_range_description >= cap2_age_group, cap2, cap)]
+  } else {
+    .sample.file[ , concessional_cap := cap]
+  }
+  
+  # https://www.ato.gov.au/uploadedFiles/Content/TPALS/downloads/Division-293-scenario-table-v2.pdf
+  .sample.file[ , excess_concessional_contributions := pmaxC(concessional_contributions - concessional_cap, 0)]
+  .sample.file[ , rental_losses := -1 * pminC(Net_rent_amt, 0)]
+  .sample.file[ , surchargeable_income_div293 := Taxable_Income + Net_fincl_invstmt_lss_amt + rental_losses + Rep_frng_ben_amt]
+  .sample.file[ , low_tax_contributions_div293 := pmaxC(concessional_contributions - excess_concessional_contributions, 0)]
+  .sample.file[ , div293_income := surchargeable_income_div293 + low_tax_contributions_div293]
+  .sample.file[ , div293_tax := ifelse(div293_income > div293_threshold, 
+                                       0.15 * pminV(low_tax_contributions_div293, 
+                                                    pmaxC(div293_income - div293_threshold, 0)), 
+                                       0)]
+  
+  # Modify taxable income to reflect exceeding cap:
+  # Note dot (too much to ask that the column wouldn't already exist)
+  .sample.file[ , .new_Taxable_Income := Taxable_Income + excess_concessional_contributions]
```

Comment:
Lets change the variable name here.  New taxable income is a bit generic, how about:

Taxable_income_for_ect